tests/e2e/unified_test.go

```diff
@@ -411,6 +411,18 @@ func runScenario(ctx context.Context, t *testing.T, testPause bool, fixture reso
 					}
 				}
 
+				if fixture.DeleteDependencies != nil {
```

Comment:
I think this can just be a randomized test.  We can run all our tests again in a different mode where we don't delete objects in order.  In this mode we will not be able to compare the golden log, but we will still be able to verify that all objects get deleted (don't get stuck)

Reply:
(And we can start with just a script that runs a few fixtures in this randomized mode)